### PR TITLE
[GH-243] docker mdl 0.9.0

### PR DESCRIPTION
**Pull request by 030**
## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Documentation (non-breaking change that does not add functionality but updates documentation)
- [x] Chore (non-breaking change that does not add functionality or fix an issue)

## Checklist:

- [x] I have read the [**CONTRIBUTING**](https://github.com/markdownlint/markdownlint/blob/master/CONTRIBUTING.md) document.
- [x] Wrote [good commit messages](https://chris.beams.io/posts/git-commit/)
- [x] Feature branch is up-to-date with `master`, if not - rebase it
- [x] Added tests for all new/changed functionality, including tests for positive and negative scenarios
- [x] The PR relates to *only* one subject with a clear title and description in grammatically correct, complete sentences

### CODE REVIEW

File: tools/docker/Dockerfile [hunk 1]
```diff
@@ -1,12 +1,10 @@
-FROM alpine:3.5
-
-MAINTAINER Mark Harrison <mark@mivok.net>
-
-RUN apk add --no-cache ruby && \
-    gem install --no-rdoc --no-ri mdl && \
+FROM alpine:3.11.3
+RUN echo "mdl:x:1001:mdl" >> /etc/group && \
+    echo "mdl:x:1001:1001:mdl,,,:/home/mdl:/sbin/nologin" >> /etc/passwd && \
+    apk add --no-cache ruby && \
```
Comment (jaymzh) [inline tools/docker/Dockerfile]: It seems like `--no-rdoc` and `--no-ri` would be desired in a docker container... why are you dropping them?

**Review by jaymzh — CHANGES_REQUESTED** [review]

File: tools/docker/Dockerfile (hunk 1 above)
Comment (030) [reply tools/docker/Dockerfile]: These options are not available anymore in recent rubygem versions:

```
  Deprecated Options:
        --[no-]rdoc                  Generate RDoc for installed gems
                                     Use --document instead
        --[no-]ri                    Generate ri data for installed gems.
                                     Use --document instead
    -u, --[no-]update-sources        Update local source cache
```

I will try to find equivalent options.

Comment (030) [reply tools/docker/Dockerfile]: According to the help menu, these options can be omitted as one has to explicitly define whether one want to install rdoc and ri documentation:

```
  Install/Update Options:
    -i, --install-dir DIR            Gem repository directory to get installed
                                     gems
    -n, --bindir DIR                 Directory where executables are
                                     located
        --document [TYPES]           Generate documentation for installed gems
                                     List the documentation types you wish to
                                     generate.  For example: rdoc,ri
```

Comment (jaymzh) [reply tools/docker/Dockerfile]: As far as I can tell from the [commit that deprecated them](https://github.com/rubygems/rubygems/commit/0d22d6025ce2653ec9fb20a4e0745c79bda677da), the default for `--document` is `ri,rdoc`. So I think you want `--no-document` (which seems to be confirmed by various blog posts like [this one](https://www.honeybadger.io/blog/how-to-globally-disable-rdoc-and-ri-during-gem-installs/).

Comment (030) [reply tools/docker/Dockerfile]: @jaymzh You are right. I have added a commit.

File: tools/docker/Dockerfile [hunk 2]
```diff
@@ -1,12 +1,11 @@
-FROM alpine:3.5
-
-MAINTAINER Mark Harrison <mark@mivok.net>
-
-RUN apk add --no-cache ruby && \
-    gem install --no-rdoc --no-ri mdl && \
+FROM alpine:3.11.3
+RUN echo "mdl:x:1000:mdl" >> /etc/group && \
+    echo "mdl:x:1000:1000:mdl,,,:/home/mdl:/sbin/nologin" >> /etc/passwd && \
+    apk add --update --no-cache \
```
Comment (jaymzh) [inline tools/docker/Dockerfile]: I just noticed, you're missing an `&&` at the end of this line.

**Review by jaymzh — CHANGES_REQUESTED** [review]

File: tools/docker/Dockerfile (hunk 2 above)
Comment (030) [reply tools/docker/Dockerfile]: It works. It is just a multi-line to improve readability:

```
docker build -t mdl .
Sending build context to Docker daemon  3.584kB
Step 1/6 : FROM alpine:3.11.3
 ---> e7d92cdc71fe
Step 2/6 : RUN echo "mdl:x:1000:mdl" >> /etc/group &&     echo "mdl:x:1000:1000:mdl,,,:/home/mdl:/sbin/nologin" >> /etc/passwd &&     apk add --update --no-cache     ruby &&     gem install mdl --no-document &&     mkdir /data
 ---> Using cache
 ---> 1529b429b910
Step 3/6 : WORKDIR /data
 ---> Using cache
 ---> b633c2f86884
Step 4/6 : USER mdl:mdl
 ---> Using cache
 ---> bb1a0d8392ef
Step 5/6 : ENTRYPOINT ["mdl"]
 ---> Using cache
 ---> 02ae0a789230
Step 6/6 : CMD ["--help"]
 ---> Using cache
 ---> 57c614f401f5
Successfully built 57c614f401f5
Successfully tagged mdl:latest
```

Comment (030) [reply tools/docker/Dockerfile]: It is fine by me to put `ruby` on the same line.

**Review by jaymzh — APPROVED** [review]

**Comment by jaymzh** [comment]
@030 - this doesn't work afaict. The version of ruby in that set is too old:

```
(1/10) Installing libffi (3.2.1-r2)
(2/10) Installing gdbm (1.12-r0)
(3/10) Installing gmp (6.1.1-r0)
(4/10) Installing ncurses-terminfo-base (6.0_p20171125-r1)
(5/10) Installing ncurses-terminfo (6.0_p20171125-r1)
(6/10) Installing ncurses-libs (6.0_p20171125-r1)
(7/10) Installing readline (6.3.008-r4)
(8/10) Installing yaml (0.1.7-r0)
(9/10) Installing ruby-libs (2.3.8-r0)
(10/10) Installing ruby (2.3.8-r0)
Executing busybox-1.25.1-r2.trigger
OK: 26 MiB in 21 packages
ERROR:  Error installing mdl:
	mixlib-config requires Ruby version >= 2.4.
```

**Comment by jaymzh** [comment]
Nope, ignore me, I hadn't updated my repo.

I'm gonna go hang my head in shame.

**Pull request merged**